src/crimson/osd/main.cc

```diff
@@ -238,6 +240,14 @@ int main(int argc, const char* argv[])
           local_conf().parse_config_files(conf_file_list).get();
           local_conf().parse_env().get();
           local_conf().parse_argv(config_proxy_args).get();
+          if (const auto ret = logfile_set_ostream(local_conf()->log_file);
+	      ret == -ENOENT) {
```

Comment:
hmm, `ENOENT` does not make sense here. can we just check for an empty `log_file` before calling this function? like
```c++
if (auto log_file = local_conf()->log_file; !log_file.empty()) {
  try {
    logfile_set_ostream(log_file);
  } catch (const std::system_error& e) {
    std::cerr << "unable to set log file: " << e.what() << std::endl;
  }
}

Reply:
IMHO, a local variable of `ofstream` would suffice.

Reply:
i thought to do that with local file, but then decided to do it just like pidfile

Reply:
could you reconsider my proposal, i just pasted my suggestion. it'd be a lot simpler if we can make it a local variable.